doc/arch/adr-002-rummager-integration-via-publishing-api.md

```diff
@@ -0,0 +1,45 @@
+# ADR 2: Rummager integration via Publishing API
+
+## Context
+
+We are currently moving to use the publishing API as the source of truth for data in our search index - we had each application integrate directly with rummager - this will help standardise how data is inserted into our index and simplify the reindexing process.
+
+Smart answers currently sends a single document to rummager per a smart answer which is indexed under the smart page URL. This document contains all the text from all the nodes (questions and outcomes) associated with the smart answer.
+
+In addition to this smart answers sends two documents to the publishing API for each smart answer:
+
+1. A document representing the start page, this document only contains the text present on the start page. This has a document type of `transaction`
+1. A document representing the flow, this has the URL for the flow base page and does not contain any content. This has a document type of `smart-answer`
+
+## Options
+
+In order to move to indexing via rummager we will need to change what data is sent to the publishing API, the following options have been considered:
+
+1. Add the content for each node to a new array item on the document schema at `details -> nodes` for the start page.
+
+  This has the negative side effect of adding the new item to all `transaction` dcument types - this document type is also used by the `publisher` application. It has also been suggested that "something feels wrong about adding context-specific data to a schema because search happens to use it".
```

Comment:
Another idea is to introduce a different schema instead of a different document type.

Schemas and document types are independent of each other, so you could capture the fact that these pages are functionally start pages, but there is smart answers specific data associated with them.